### PR TITLE
Add value exception for UnsupportedTemporalUnit

### PR DESCRIPTION
**Pull request by Lojjs**

### CODE REVIEW

File: community/values/src/main/java/org/neo4j/values/storable/LocalDateTimeValue.java [hunk 1]
```diff
@@ -317,13 +318,13 @@ ZoneId getZoneId( Supplier<ZoneId> defaultZone )
     @Override
     ZoneId getZoneId()
     {
-        throw new UnsupportedTemporalTypeException( "Cannot get the timezone of" + this );
+        throw new UnsupportedTemporalUnitException( "Cannot get the timezone of: %s" + this );
```
Comment (sherfert) [inline community/values/src/main/java/org/neo4j/values/storable/LocalDateTimeValue.java]: Don't you need a `String.format` with the `%s` ?

File: community/values/src/main/java/org/neo4j/values/storable/LocalTimeValue.java [hunk 2]
```diff
@@ -108,7 +109,16 @@ public static LocalTimeValue truncate(
...
         LocalTime localTime = input.getLocalTimePart();
-        LocalTime truncatedLT = localTime.truncatedTo( unit );
+        LocalTime truncatedLT;
+        try
+        {
+            truncatedLT = localTime.truncatedTo( unit );
+        }
+        catch ( UnsupportedTemporalTypeException e )
+        {
+            throw new UnsupportedTemporalUnitException( e.getMessage() );
```
Comment (sherfert) [inline community/values/src/main/java/org/neo4j/values/storable/LocalTimeValue.java]: You should always pass in `e` as the cause (and thus make your Exception types accept an optional cause).

File: community/values/src/main/java/org/neo4j/values/utils/UnsupportedTemporalUnitException.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values.utils;
+
+
+public class UnsupportedTemporalUnitException extends ValuesException
+{
+    public UnsupportedTemporalUnitException( String errorMsg )
```
Comment (sherfert) [inline community/values/src/main/java/org/neo4j/values/utils/UnsupportedTemporalUnitException.java:25]: This should have 2 constructors: One with cause and this one you already have.

File: community/cypher/runtime-util/src/main/scala/org/neo4j/cypher/exceptionHandler.scala [hunk 4]
```diff
@@ -101,4 +105,11 @@ object exceptionHandler extends MapToPublicExceptions[CypherException] {
   trait RunSafely {
     def apply[T](body: => T)(implicit f: ExceptionHandler = ExceptionHandler.default): T
   }
+
+  def mapToCypher(exception: ValuesException): CypherException = {
+    exception match {
+      case e: UnsupportedTemporalUnitException =>
+        exceptionHandler.cypherTypeException(e.getMessage, e.getCause)
```
Comment (sherfert) [inline community/cypher/runtime-util/src/main/scala/org/neo4j/cypher/exceptionHandler.scala]: `e.getCause` will be `null`. You never set it. Also, I think you should use `e` instead.

**Review by sherfert — CHANGES_REQUESTED** [review]
In general the architecture of the changes looks good to me and should be applied the same way for the other Exception types.

There are some small issues, though.

File: community/values/src/main/java/org/neo4j/values/storable/LocalDateTimeValue.java (hunk 1 above)
Comment (Lojjs) [reply community/values/src/main/java/org/neo4j/values/storable/LocalDateTimeValue.java]: Yes, I do. And I have missed it in more places...

**Review by sherfert — APPROVED** [review]

**Pull request merged**